pysc2/agents/scripted_agent.py

```diff
@@ -49,24 +49,29 @@ def step(self, obs):
 
 class CollectMineralShards(base_agent.BaseAgent):
   """An agent specifically for solving the CollectMineralShards map."""
+  current_player = 0
 
   def step(self, obs):
     super(CollectMineralShards, self).step(obs)
-    if FUNCTIONS.Move_screen.id in obs.observation["available_actions"]:
-      player_relative = obs.observation["feature_screen"][_PLAYER_RELATIVE]
-      neutral_y, neutral_x = (player_relative == _PLAYER_NEUTRAL).nonzero()
-      player_y, player_x = (player_relative == _PLAYER_SELF).nonzero()
-      if not neutral_y.any() or not player_y.any():
+    player_units = list(filter(lambda unit: unit[1] == _PLAYER_SELF, obs.observation['feature_units']))
```

Comment:
Verify that the feature_units observation exists, and give a decent error message if it's not.